src/timer.rs

```diff
@@ -0,0 +1,27 @@
+use std::time::{Duration, Instant};
+pub struct Timer {
+	period: Duration,
+	init: Instant,
+}
+impl Timer {
+	pub fn time_per_second(times: f32) -> Timer {
+		Timer::new(Duration::from_secs_f32(1.0 / times))
+	}
+	pub fn new(period: Duration) -> Timer {
+		Timer {
+			period,
+			init: Instant::now(),
+		}
+	}
+	pub fn tick(&mut self) -> bool {
+		if self.init.elapsed() >= self.period {
+			self.init = Instant::now();
+			true
+		} else {
+			false
+		}
+	}
+	pub fn ms_remaining(&self) -> Duration {
+		self.period - self.init.elapsed()
+	}
```

Comment:
Unlike my API draft, this isn't `ms_remaining`, it's any time value the user wants to turn the duration into. Probably call this `remaining` or `time_remaining`.

Additionally, what's the behavior when `elapsed` > `period`? My instinct says this function should never return negative time, but I'm not sure what even happens in that case.

Reply:
good catch upon the name. I, should've changed that when I changed the return type.

About the elapsed > period case. Seems the code just calls `checked_sub().expect()` so this is defiantly something that needs to change. The easiest would be to match the return type of `checked_sub` and return an `Option<Duration>`.